homeassistant/components/template/light.py

```diff
@@ -328,24 +423,42 @@ async def async_turn_on(self, **kwargs):
             self._temperature = kwargs[ATTR_COLOR_TEMP]
             optimistic_set = True
 
-        if ATTR_BRIGHTNESS in kwargs and self._level_script:
-            await self._level_script.async_run(
-                {"brightness": kwargs[ATTR_BRIGHTNESS]}, context=self._context
-            )
-        elif ATTR_COLOR_TEMP in kwargs and self._temperature_script:
+        common_params = {}
+
+        if ATTR_BRIGHTNESS in kwargs:
+            common_params["brightness"] = kwargs[ATTR_BRIGHTNESS]
+
+        if ATTR_TRANSITION in kwargs:
+            common_params["transition"] = kwargs[ATTR_TRANSITION]
+
+        if ATTR_COLOR_TEMP in kwargs and self._temperature_script:
+            common_params["color_temp"] = kwargs[ATTR_COLOR_TEMP]
+
             await self._temperature_script.async_run(
-                {"color_temp": kwargs[ATTR_COLOR_TEMP]}, context=self._context
+                common_params, context=self._context
             )
         elif ATTR_WHITE_VALUE in kwargs and self._white_value_script:
+            common_params["white_value"] = kwargs[ATTR_WHITE_VALUE]
+
             await self._white_value_script.async_run(
-                {"white_value": kwargs[ATTR_WHITE_VALUE]}, context=self._context
+                common_params, context=self._context
             )
+        elif ATTR_EFFECT in kwargs and self._effect_script:
+            common_params["effect"] = kwargs[ATTR_EFFECT]
+
+            await self._effect_script.async_run(common_params, context=self._context)
         elif ATTR_HS_COLOR in kwargs and self._color_script:
             hs_value = kwargs[ATTR_HS_COLOR]
+            common_params["hs"] = hs_value
+            common_params["h"] = int(hs_value[0])
+            common_params["s"] = int(hs_value[1])
+
             await self._color_script.async_run(
-                {"hs": hs_value, "h": int(hs_value[0]), "s": int(hs_value[1])},
+                common_params,
                 context=self._context,
             )
+        elif ATTR_BRIGHTNESS in kwargs and self._level_script:
```

Comment:
Can it be considered a breaking change to move brightness to the lowest priority?
Also, the behavior where only a single script is executed and the priority of the scripts needs to be documented.

Reply:
Yes and I don't think it was correct behavior

Reply:
also added a similar transition support